### ISSUE TITLE
Using Omit weakens type checking

### ISSUE THREAD

**Issue by Bastian** [microsoft/TypeScript#46361]
# Bug Report

When using the `Omit` type, the type-checking does not catch some cases that it previously did.

### 🔎 Search Terms

Omit inaccurate

### 🕗 Version & Regression Information

Test in both 3.3.3 and 4.5.0-beta.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.5.0-beta#code/JYOwLgpgTgZghgYwgAgILIN4ChnLATwAcIAuZAIjnIG4dlC58AbAezgBMyBnMKUAc1q4uLALYQwACwHdeA2gF8sWUJFiIUAIUx0CxMuQBGNOg2ZtOyQyxZMIcEEOQjxUmc7khBWJVj0pUbQBeNGQAH2RNWj8iAM0Ad2ApFgBXMABlMQlpL2QQgHlRJIAeQIAaChdsgXIAPmiEFhAeZDgARjJAxOS0zNcc-jydXH8DKjLTRlYOMl4UiAqAekXkABVpLmdJVKZ2ZBAWMGR4ligAawrDNOQk5HYWCE2AChSQCAAPYgRIdgBKHyEWEazSOcAATJ1gsM8LExuQJrgzNNLHMFshlsgAOpJbbXKQoQpJCr45CNUSEYB2KCksSEJhwUCbSTQFBPD5fH7-YRZNxeAzwgFAA)

### 💻 Code

```ts
interface A {
  type: "a";
  payload: string;
  something: string;
}

interface B {
  type: "b";
  payload: boolean;
  something: string;
}

type AB = A | B;

type ABwithoutSomething = Omit<AB, "something">;

const a1: ABwithoutSomething = {
  type: "a",
  payload: true, // This should not work, but it does (unexpected)
};

const a2: AB = {
  type: "a",
  payload: true, // Without the Omit, the compiler complains here (expected)
  something: "",
};

```

### 🙁 Actual behavior

When using the `Omit` type to exclude a single field, the type checking is weaker.

### 🙂 Expected behavior

When using `Omit`, the only difference should be that the field is excluded without any other side-effects or weaker typechecks.
